core/lib/services/writables.spec.ts

```diff
@@ -0,0 +1,47 @@
+import {writable} from '@amadeus-it-group/tansu';
+import type {SpyInstance} from 'vitest';
+import {beforeEach, describe, expect, test, vi} from 'vitest';
+import {writableWithDefault} from './stores';
+import {typeArray} from './writables';
+
+describe(`Writables service`, () => {
+	let consoleErrorSpy: SpyInstance<Parameters<typeof console.error>, ReturnType<typeof console.error>>;
+
+	beforeEach(() => {
+		consoleErrorSpy = vi.spyOn(console, 'error').mockImplementation(() => {});
+	});
+
+	test(`should support typeArray`, () => {
+		const config$ = writable([0]);
+		const store$ = writableWithDefault<number[]>([0], config$, typeArray);
+		// set the wrong value
+		store$.set(100 as any);
+		expect(consoleErrorSpy).toHaveBeenCalledWith('Not setting invalid value', 100);
+		// the store is not updated with the wrong value
+		expect(store$()).toStrictEqual([0]);
+
+		const testArray = [15, 20];
+		store$.set(testArray);
+
+		expect(store$()).toStrictEqual([15, 20]);
+
+		testArray[1] = 15;
```

Comment:
I am not really in favor of using stores this way. I prefer immutability, otherwise we need `equal` functions that consider objects as always different. I would remove this part of the test.